src/dom-utils.ts

```diff
@@ -27,55 +27,77 @@ export function moveFocusToDialog(el: HTMLElement) {
   focused.focus()
 }
 
-/**
- * Get focusable children by recursively traversing the subtree of the given
- * element, while accounting for Shadow DOM subtrees
- */
-export function getFocusableChildren(el: ParentNode): HTMLElement[] {
-  // Check for the base case of our recursion
-  if (el instanceof HTMLElement) {
-    // If this element is marked as inert, neither it nor any member of its
-    // subtree will be focusable, therefore there are no focusable children
-    if (el.inert) return []
+// Get the first and last focusable elements in a given subtree
+export function getFirstAndLastFocusableChild(el: HTMLElement) {
+  const first = findFocusableElement(el, true)
+  let last = null
+  if (first !== null) {
+    last = findFocusableElement(el, false) || first
+  }
+  return [first, last]
+}
 
-    // If this element has no children or if it nullifies its children’s
-    // semantics, there are no focusable children
-    if (!el.firstElementChild || el.matches(PRESENTATIONAL_CHILDREN_SELECTOR)) {
-      // If the element itself is focusable, return it otherwise return nothing
-      return isFocusable(el) ? [el] : []
-    }
+function findFocusableElement(
+  node: HTMLElement,
+  forward: boolean
+): HTMLElement | null {
+  // If we're walking forward, check if this node is focusable,
+  // and return it immediately if it is.
+  if (forward && isFocusable(node)) {
+    return node
   }
```

Comment:
This means that it returns itself if the element is focusable, similar to how `closest` returns itself if the element matches. Is that correct?

Reply:
That's correct!